crypto/crypto.go

```diff
@@ -20,8 +20,30 @@ func GetPubKeyHex(privateKey PrivateKey) string {
 	return hex.EncodeToString(privKey.PublicKey.X) + hex.EncodeToString(privKey.PublicKey.Y)
 }
 
-func GeneratePrivateKey() (PrivateKey, error) {
-	return elliptic.GeneratePrivateKey(elliptic.Secp256k1)
+func GeneratePrivateKey(difficulty int) (PrivateKey, error) {
+	for true {
+		privKey, err := elliptic.GeneratePrivateKey(elliptic.Secp256k1)
```

Comment:
check for `err` and don't proceed if it's not null

Reply:
done